frontend/src/scenes/actions/actionEditLogic.js

```diff
@@ -0,0 +1,85 @@
+import { kea } from 'kea'
+import api from 'lib/api'
+import { uuid } from 'lib/utils'
+import { toast } from 'react-toastify'
+
+export const actionEditLogic = kea({
+    key: props => props.id || 'new',
+    actions: () => ({
+        saveAction: true,
+        setAction: action => ({ action }),
+        setCreateNew: createNew => ({ createNew }),
+        setErrorActionId: actionId => ({ actionId }),
+    }),
+
+    loaders: ({ props }) => ({
+        action: {
+            loadAction: async () => {
+                return await api.get(props.apiURL + 'api/action/' + props.id)
+            },
+        },
+    }),
+
+    reducers: () => ({
+        action: [
+            null,
+            {
+                setAction: (_, { action }) => action,
+            },
+        ],
+        errorActionId: [
+            null,
+            {
+                setErrorActionId: (_, { actionId }) => actionId,
+            },
+        ],
+        createNew: [
+            false,
+            {
+                setCreateNew: (_, { createNew }) => createNew,
+            },
+        ],
+    }),
+
+    listeners: ({ values, props, actions }) => ({
+        saveAction: async () => {
+            let action = { ...values.action }
+            actions.setErrorActionId(null)
+            action.steps = action.steps.map(step => {
+                if (step.event == '$pageview') step.selection = ['url', 'url_matching']
```

Comment:
Good to do `let action = { ...values.action }` above, yet now both objects (`action` and `values.action`) share the same array with the same elements. Calling `action.steps = action.steps.map` is great for making a new `steps` array, yet each `step` inside both arrays still points to the same object in memory. You should add another `const localstep = {...actionStep}` I guess.